### ISSUE TITLE
Muse reader treats `;` not in the first column as comment

### ISSUE THREAD

**Issue by jgm** [jgm/pandoc#4551]
https://travis-ci.org/jgm/pandoc/jobs/366652412

```
  Blocks
        Round trip:                                                FAIL (1.14s)
          *** Failed! Falsifiable (after 72 tests):
          DefinitionList [([Str "-",Str "",Space],[[Header 6 ("",[],[]) [Str "1A.yow",Str "T^X",Space],Div ("",[],[("a","11"),("b_2","a b c")]) [Plain [RawInline (Format "html") "<a id=\"eek\">"],HorizontalRule,Para [Str "!l",RawInline (Format "latex") "\\my{command}",Space]]]]),([Str "s)1J",Str "\9437jXNdM",Str "4)w%bJ"],[[Para [Str "",Space,Str ";"],LineBlock [[Str ",w",Str "",Code ("loc",["c","numberLines"],[]) "uS",Str "x.\4235B\7048o"]]],[Para [Str "P(G",Space],LineBlock [[Str "",Str "Pd<\4864\5428!A6",Str "_\6454"]],Para [Str "(\675{h8=rL",Str "G&\4401]4"]]]),([Str ">\8411t",Str ""],[[CodeBlock ("",["c","numberLines"],[("start","22")]) "K"],[BlockQuote [Para [RawInline (Format "html") "<a id=\"eek\">",RawInline (Format "latex") "\\my{command}"],Plain [RawInline (Format "latex") "\\my{command}",Space,Str "G;y.R"],CodeBlock ("",["c","numberLines"],[]) ",#&\4590\""]],[Para [Str "eq%(]",Str ")R"],DefinitionList [([Str "\DELp'7u\7381",Str "0KLzo3-"],[[Para [Str ";",Code ("",["c","numberLines"],[]) "",Space],CodeBlock ("loc",["c","numberLines"],[]) "Q\DEL",Plain [Str "%\8889\&5s"]],[Para [Str "qnY'\3394'K"],Plain [Str "mjSHH",Str "O",Str "oSgaU;"],Para [Str "*)\\r\9550w6",Space,Space]],[Plain [Code ("",[],[("start","22")]) "-|\4572ZwV",Str "/",Space,Space],CodeBlock ("loc",["haskell"],[]) "&N",Plain [Code ("",["c","numberLines"],[("start","22")]) "Z[{3WXu",Space]]]),([Str "",Str "WASR\2615",Str "|f"],[[Para [Str "_",Space,Str "Av:&zk",Str "WI,\3417*"],Plain [Str "9j?(T",Str "D",Str "",Code ("",["c","numberLines"],[]) "g'\3586u"]],[Plain [Str "",Space,Space],Plain [Str "ndM",Space],LineBlock [[Quoted SingleQuote [Str "0b)Ck\"Z"],Link ("",["haskell"],[]) [Str "\DEL~",Space] ("yvC","$|\9257Tbr8"),Quoted DoubleQuote [Str "f2e+#9Qi",Space,Str "\192Y\9401q5"],SmallCaps [Str "E",RawInline (Format "latex") "\\my{command}",Str "_vu:~Y \1910",Space]]]],[LineBlock [[Str "m [\5421)g",Superscript [Str "KCe\7238\\",Space]]],CodeBlock ("",["haskell"],[]) "<$`",Para [Str "$_",RawInline (Format "html") "<a id=\"eek\">",Code ("loc",["c","numberLines"],[("start","22")]) "U\8869\7961\&3}",Str "",Str "Hk9\9584"]]]),([Str "r",Space,Space],[[RawBlock (Format "latex") "\\begin[opt]{env}\nhi\n{\\end{env}"],[CodeBlock ("",["c","numberLines"],[]) "|B-P2",Plain [Code ("",["c","numberLines"],[]) "~Ii",Space]]])],BlockQuote [Para [Str "P<)90^"]]]])]
```

@labdsf - these keep coming up.  Why don't you remove the round-trip tests from the test suite, and create a separate program you can use to round-trip muse intensively?  These non-deterministic failures are not good.

**Comment by labdsf** [comment]
Minimal example:
```
 foo :: ; bar
```

`;` is treated as comment, need to add `guard`.

**Comment by labdsf** [comment]
> @labdsf - these keep coming up. Why don't you remove the round-trip tests from the test suite, and create a separate program you can use to round-trip muse intensively?

The failures are very rare now. I am running
```
$ while stack test --test-arguments='-p Round'; do echo ok; done
```
and it does not generate failures for hundreds of runs (that is, tens of thousands of round-trip tests).

The last "wave" of bugs was caused by a change in round-trip test, I fixed most bugs since then.

The bug just found is a very rare case, caused by a string of single ";" following `Space` and `Str ""`. And it is a good thing it was found, I just fixed it.

I would like to keep this test, because 100 tests it runs catch bugs in new functionality very fast if something is really broken. Even if development of Muse slows down (because it is almost complete already), it will catch bugs in shared code, regressions in upstream libraries (like TagSoup) etc.

I think I will run round-trip tests for several hours more just to make sure the failures are really-really rare (0 failures is the only acceptable outcome) and keep the test in the test suite after that, ok?
